scripts/asciidoc2confluence.groovy

```diff
@@ -386,20 +438,26 @@ def pushToConfluence = { pageTitle, pageBody, parentId, anchors, pageAnchors ->
 
     def pages
     trythis {
-        def cql = "space='${confluenceSpaceKey}' AND type=page AND title~'" + realTitle(pageTitle) + "'"
+        // #342-dierk42: Colons in title irritate Confluence's lucene search engine
+        // (not really part of #342 but useful)
+        def sPageTitle = pageTitle.replace(':', '')
```

Comment:
hm. shouldn't we escape the ':' as special character? And aren't there more such special characters?

Reply:
Agreed. The colon was a quick hack for me. As we are searching for titles all characters with a special meaning should be escaped/deleted/masked. This is valid for colon, the relational operators like '<', '!=' and so on. An of course the Wildcard characters '*' and '?'.

As your using the CONTAINS-operator in the cql (title~...) it should be sufficient to delete all these characters. The rest are words or numbers that are ok. 

I have no experience what happens when you use the '=' for the title search. Perhaps then you'ld need the special characters? 

To cope with these special characters would make the routine more robust I think.

Reply:
The other question regarding adding documentation: Of course I can do it. Where in the docs would you like to have it?
Some documentation is needed because Confluence allows no spaces in the keywords. One schould know this fact. But something lile "this_is_a_label" would be correct.

We could improve the part where the list of keywords in asciidoc2confluence.groovy is built up and replace all whitespace by '_', after the trim operation. (line 612 ff in my version)

Reply:
regarding the special characters: I just did a quick google search and there seem to be some open bugs with this: https://jira.atlassian.com/browse/CONFSERVER-55559
...and I remember that we already have some fixes in place like chacking for the number of search results.
So let's leave it as it is for the moment...

Reply:
regarding the docs: take a look at https://doctoolchain.github.io/docToolchain/#_publishtoconfluence
there is an orange button in the upper right corner which will take you to the sources...
Any documentation is appreaciated!